### ISSUE TITLE
Undefined values ???

### ISSUE THREAD

**Issue by gpicron** [ssbc/bipf#43]
The library encode undefined as value 2 for BOOLNULL tag while nothing defined in spec.

Note that seems a deviation from JSON compatibility and is not very portable as differentiation between undefined and null does not exist in most language.

**Comment by staltz** [comment]
I agree it's weird, and probably not very necessary in software that currently uses BIPF.

**Comment by gpicron** [comment]
I made change proposal on the spec https://github.com/ssbc/bipf-spec/pull/3.  In practice, the side effect is that "undefined" is compliant...

**Comment by gpicron** [comment]
Any plan to resolve ? There must be reason that you introduced that. In Manyverse ? In ssb stack components ? If I can help fixing it, just tell me. And drive me towards the component that depends on it and I can propose alternative to avoid it

**Comment by staltz** [comment]
If I remember correctly (and I don't always do) it was not my choice to included undefined. BIPF was originally written by Dominic Tarr and there's few modifications that arj and I made in its core logic.

**Comment by staltz** [comment]
https://github.com/ssbc/bipf/commit/fc226e8fc9536af243adf5c97d606239e0c79dac

**Comment by gpicron** [comment]
Yes I found it too...  Will be hard determine what it can impact to realign to the spec...
Note, in parallel, I found that in a base stack setup with ssb-db2 , there are some message with such path having a undefined value...   Maybe I will be able to find out where come from by tracing back.

**Comment by staltz** [comment]
Yeah I bet there are malformed SSB messages with `undefined` in them. Not by design, probably a mistake. But we still don't want the application to crash upon reading those. At least coercing them to null.
